scripts/gen_dockerfile_test.py

```diff
@@ -0,0 +1,228 @@
+#!/usr/bin/env python3
+
+# Copyright 2017 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit test for gen_dockerfile.py"""
+
+import argparse
+import os
+import re
+import shutil
+import subprocess
+import tempfile
+import unittest
+import unittest.mock
+
+import yaml
+
+import gen_dockerfile
+
+# Expected list of files generated
+EXPECTED_OUTPUT_FILES = ['Dockerfile', '.dockerignore']
+
+
+class GenDockerfileTest(unittest.TestCase):
```

Comment:
While I respect that you work almost exclusively within google infrastructure, I want to briefly express my opinion as an external developer (and contributor). Please do not take this personally, but I am going to strongly challenge your answer here.

I think that "I'm doing it this way because it works inside of Google" is an awfully anti-collaborative attitude to have. We do so many weird and frankly outdated things within Google that it gives the impression to Python developers that we're still behind the times. I think it is so critically important for us to be of the Python community first and of Google second. Regardless of the merits of pytest and unittest, I think this is a dangerous attitude to have and I would really hate to see it permeate into other projects. I think adopting modern tooling, even if it's slightly unfamiliar and inconvenient to us at first, is important in our projects being attractive to the broader Python community.

Now, on to unittest vs pytest: unittest follows the standard library philosophy of being useful, but not necessarily the best tool for the job. Just like you wouldn't use urllib over requests, I think it's silly to avoid pytest. pytest outclasses unittest in every way - from the test runner, to assertions, to plugins, to the framework, to interoperability with unittest and nose, and to the overall community. This test file itself would be significantly easier to read if you used pytest - pytest has no required API. Oh - and if you used tox or nox you'd still be able to run them super easily in one command on your workstation. There's been [lots](http://halfcooked.com/presentations/pyconau2013/why_I_use_pytest.html) of [great](http://thread.gmane.org/gmane.comp.python.testing.general/3748) [writeups](http://pythontesting.net/framework/pytest/pytest-introduction/) about pytest.

If you're still not convinced, that's fine. You're the owner of this project. I am just paid to speak my mind, so I am doing that.